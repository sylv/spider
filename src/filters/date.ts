import type { Filter } from "src/schema.js";
import { parseAbsoluteDate } from "@atlasbot/parsers";

export const dateFilter: Filter<Date> = (input) => {
  const result = parseAbsoluteDate(input)
  if (!result) {
    throw new Error(`Expected a date string, got "${input}"`)
  }

  return result;
};

if (import.meta.vitest) {
  const { it, expect } = import.meta.vitest;
  it("should parse valid date strings", () => {
    expect(dateFilter("2022-01-01")).toMatchSnapshot();
    expect(dateFilter("2022-02-01T00:00:00.000Z")).toMatchSnapshot();
    expect(dateFilter("21st dec 2021")).toMatchSnapshot();
    expect(dateFilter("dec 21 2023")).toMatchSnapshot();
  });

  it("should throw an error if input is not a valid date string", () => {
    expect(() => dateFilter("not a date")).toThrow();
  });
}
