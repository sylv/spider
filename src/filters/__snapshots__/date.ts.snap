// Vitest Snapshot v1

exports[`should parse valid date strings 1`] = `2022-01-01T04:00:00.000Z`;

exports[`should parse valid date strings 2`] = `2022-02-01T00:00:00.000Z`;

exports[`should parse valid date strings 3`] = `2021-12-21T04:00:00.000Z`;

exports[`should parse valid date strings 4`] = `2023-12-21T04:00:00.000Z`;
